internal/core/application/trade_service.go

```diff
@@ -445,40 +445,47 @@ func (t *tradeService) tradeComplete(
 ) (txID string, swapFail domain.SwapFail, err error) {
 	trade, err := t.tradeRepository.GetTradeBySwapAcceptID(ctx, swapComplete.GetAcceptId())
 	if err != nil {
-		return "", nil, err
+		return
 	}
 
 	tradeID := trade.ID
-	err = t.tradeRepository.UpdateTrade(
-		ctx,
-		&tradeID,
-		func(trade *domain.Trade) (*domain.Trade, error) {
-			psetBase64 := swapComplete.GetTransaction()
-			res, err := trade.Complete(psetBase64)
-			if err != nil {
-				return nil, err
-			}
-			if !res.OK {
-				swapFail = trade.SwapFailMessage()
-				return trade, nil
-			}
 
-			log.Info("trade with id ", tradeID, " completed")
+	psetBase64 := swapComplete.GetTransaction()
+	res, err := trade.Complete(psetBase64)
+	if err != nil {
+		return
+	}
 
-			if _, err := t.explorerSvc.BroadcastTransaction(res.TxHex); err != nil {
-				return nil, err
-			}
+	if !res.OK {
+		swapFail = trade.SwapFailMessage()
+		return
+	}
+	log.Info("trade with id ", tradeID, " completed")
 
-			txID = res.TxID
+	if _, err = t.explorerSvc.BroadcastTransaction(res.TxHex); err != nil {
+		return
+	}
 
-			log.Info("trade with id ", tradeID, " broadcasted: ", txID)
+	txID = res.TxID
+	log.Info("trade with id ", tradeID, " broadcasted: ", txID)
+
+	go t.persistTradeCompleted(trade)
```

Comment:
we just updated trade here, maybe it would be cleaner if we handle this in anonymous func, than having new method called persistTradeCompleted which is actually only updates trade.

Reply:
done